python-backend/tests/applications/resources/test_application_resource.py

```diff
@@ -0,0 +1,133 @@
+    db.session.commit()
+    db.session.delete(mine_1)
+    db.session.delete(mine_2)
+    db.session.commit()
+
+
+# GET
+def test_get_application_not_found(test_client, setup_info, auth_headers):
+    get_resp = test_client.get(
+        '/applications/' + setup_info.get('bad_guid'), headers=auth_headers['full_auth_header'])
+    get_data = json.loads(get_resp.data.decode())
+    assert get_data == {'error': {'status': 404, 'message': 'Application not found'}}
+    assert get_resp.status_code == 404
+
+
+def test_get_application(test_client, setup_info, auth_headers):
+    get_resp = test_client.get(
+        '/applications/' + setup_info.get('mine_1_application_guid'),
+        headers=auth_headers['full_auth_header'])
+    get_data = json.loads(get_resp.data.decode())
+    assert get_data['application_guid'] == setup_info.get('mine_1_application_guid')
+    assert get_resp.status_code == 200
+
+
+def test_get_applications_on_a_mine(test_client, setup_info, auth_headers):
+    get_resp = test_client.get(
+        '/applications?mine_guid=' + setup_info.get('mine_1_guid'),
+        headers=auth_headers['full_auth_header'])
+    get_data = json.loads(get_resp.data.decode())
+    assert get_resp.status_code == 200
+    assert len(get_data) == 1
+
+
+#Create
+def test_post_application(test_client, setup_info, auth_headers):
+    APP_NO = 'TX-54321'
+    data = {
+        'mine_guid': setup_info.get('mine_2_guid'),
+        'application_no': APP_NO,
+        'application_status_code': 'RIP',
+        'description': 'This is a test.',
+        'received_date': '1999-12-12',
+    }
+    post_resp = test_client.post(
+        '/applications', headers=auth_headers['full_auth_header'], data=data)
+    post_data = json.loads(post_resp.data.decode())
+
+    assert post_resp.status_code == 200
+    assert len(
+        [key for key, value in post_data.items() if key not in data or value != data[key]]) == 2
+
+
+def test_post_application_bad_mine_guid(test_client, setup_info, auth_headers):
+    data = {'mine_guid': setup_info.get('bad_guid')}
+    post_resp = test_client.post(
+        '/applications', headers=auth_headers['full_auth_header'], data=data)
+
+    assert post_resp.status_code == 404
+
+
+def test_post_with_permit_guid(test_client, setup_info, auth_headers):
+    APP_NO = 'TX-54321'
+    data = {
+        'mine_guid': setup_info.get('mine_2_guid'),
+        'application_no': APP_NO,
+        'application_status_code': 'RIP',
+        'description': 'This is a test.',
+        'received_date': '1999-12-12',
+    }
+    post_resp = test_client.post(
+        '/applications/' + setup_info.get('mine_1_application_guid'),
+        headers=auth_headers['full_auth_header'],
+        data=data)
+    assert post_resp.status_code == 400
+
+
+#Put
+def test_put_application(test_client, setup_info, auth_headers):
+    application_guid = setup_info.get('mine_1_application_guid')
+    application = Application.find_by_application_guid(application_guid)
```

Comment:
Is there a reason not to yield the application object (instead of just it's guid. I have been doing that in other tests.

Reply:
I did it this way because the object can change from when it was created. I know its not very likely the guid would change but just in case id rather grab a instance of the object.